src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,963 @@
+        // map events
+        fluid.orator.domReader.mapUtteranceEvents(that, options, that.options.utteranceEventMap);
+
+        return fluid.promise.fireTransformEvent(that.events.onQueueSpeech, text, options);
+    };
+
+    /**
+     * Unwraps the contents of the element by removing the tag surrounding the content and placing the content
+     * as a node within the element's parent. The parent is also normalized to combine any adjacent textnodes.
+     *
+     * @param {String|jQuery|DomElement} elm - element to unwrap
+     */
+    fluid.orator.domReader.unWrap = function (elm) {
+        elm = $(elm);
+
+        if (elm.length) {
+            var parent = elm.parent();
+            // Remove the element, but place its contents within the parent.
+            elm.contents().unwrap();
+            // Normalize the parent to cleanup textnodes
+            parent[0].normalize();
+        }
+    };
+
+    /**
+     * Sets the parseQueue and related model values
+     *
+     * @param {Component} that - the component
+     * @param {DomWordMappings} parseQueue - An array of {DomWordMap} objects containing the position mappings from a parsed
+     *                                   {DomElement}.
+     */
+    fluid.orator.domReader.setParseQueue = function (that, parseQueue) {
+        that.parseQueue = parseQueue || [];
+        that.applier.change("", {
+            parseQueuelength: that.parseQueue.length,
+            parseIndex: null,
+            ttsBoundary: null
+        }, "ADD", "resetParseQueue");
+    };
+
+    /**
+     * Positional information about a word parsed from the text in a {DomElement}. This can be used for mappings between
+     * a synthesizer's speech boundary and the word's location within the DOM.
+     * The structure of each data point is as follows:
+     *  {
+     *      blockIndex: {Integer}, // the index into the entire block of text being parsed from the DOM
+     *      startOffset: {Integer}, // the start offset of the current `word` relative to the closest
+     *                             // enclosing DOM element
+     *      endOffset: {Integer}, // the end offset of the current `word` relative to the closest
+     *                           // enclosing DOM element
+     *      node: {DomNode}, // the current child node being parsed
+     *      childIndex: {Integer}, // the index of the child node being parsed relative to its parent
+     *      parentNode: {DomElement}, // the parent DOM node
+     *      word: {String} // the text, `word`, parsed from the node. (It may contain only whitespace.)
+     *   }
+     *
+     * @typedef {Object} DomWordMap
+     */
+
+    /**
+     * An array of {DomWordMap} objects detailing the words parsed from the a {DomElement}. This information can be used
+     * to map the synthesized speech to the location of each word in the DOM; which is useful for highlighting words as
+     * they are being read.
+     *
+     * @typedef {Array} DomWordMappings
```

Comment:
On the one hand I don't think this needs to be a typedef since it is just an array, and on the other hand I think it could usefully be documented what it is actually an array of - that is, you should use {DomWordMap[]} as the @param type below